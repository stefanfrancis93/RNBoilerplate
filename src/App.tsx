import Root from 'components/Root';
import React from 'react';
import { Provider as PaperProvider } from 'react-native-paper';
import { Provider } from 'react-redux';

import configureStore from './store/configureStore';

const App: () => JSX.Element = () => {
  const store = configureStore();

  return (
    <Provider store={store}>
      <PaperProvider>
        <Root />
      </PaperProvider>
    </Provider>
  );
};

export default App;
