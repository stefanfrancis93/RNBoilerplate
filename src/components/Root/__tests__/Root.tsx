import App from '@components/Root';
import React from 'react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const component = renderer.create(<App />).toJSON();
  expect(component).toMatchSnapshot();
});
