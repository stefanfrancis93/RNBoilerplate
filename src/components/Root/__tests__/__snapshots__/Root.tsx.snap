// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityStates=""
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "rgba(0, 0, 0, 0.29)",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0.5,
        "elevation": 0,
        "minWidth": 64,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 12,
              "marginRight": -4,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Array [
                Object {
                  "color": "#6200ee",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û∞çÉ
          </Text>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</View>
`;
